src/Template/Sentences/show.ctp

```diff
@@ -150,14 +148,25 @@ echo $this->element('/sentences/navigation', [
         }
 
         if (!empty($contributions)) {
+            $active_links = array();
             echo '<md-list id="logs">';
             foreach ($contributions as $contribution) {
                 echo $this->element(
                     'logs/log_entry_annexe',
                     array('log' => $contribution)
                 );
+                if ($contribution->type == "link" and $contribution->action == "insert") {
+                    $active_links[$contribution->translation_id] = 1;
+                }
+                if ($contribution->type == "link" and $contribution->action == "delete") {
+                    unset($active_links[$contribution->translation_id]);
+                }
             }
             echo '</md-list>';
+            if (!empty($active_links)) {
+                $this->Html->script(['jquery.scrollTo.min.js', 'sentences.logs.js'],
+                                    ['block' => 'scriptBottom']);
+            }
```

Comment:
A more simple way would be to just check if `$sentence->translations` is empty. Analyzing the logs works just fine too, but it's a bit more lines of code :)

Now that doesn't take into account that logs are not 100% reliable (#1653). And if you would want to be completely accurate about when to load the javascript, you could keep your code above and additionally check if the translations in the `$active_links` actually exist in `$sentence->translations`.

But if you ask me, we don't have to go into that level of details. Such cases would be rare and we don't have to over-complexify the code for a low impact optimization.

Reply:
Yeah, after sleeping over it, I realized that I was overthinking the problem. :-)

But while testing for a sentence without a log (by manually inserting it into the sentences table using the mysql REPL) I've noticed that the condition on line 152 doesn't work: https://github.com/Tatoeba/tatoeba2/blob/a171662ded228c0e5be38a56b6504ea009f68e38/src/Template/Sentences/show.ctp#L152

I've expected to get the message "There is no log for this sentence" from the else branch. But since `$contributions` is a CakePHP ResultSet it will never be empty because an empty ResultSet is an object containing an empty list. Thus I think a better way for testing whether `$contributions` is empty would be to check its count.

I've updated the PR (hopefully the last time :-) )